modules/helfi_node_announcement/modules/helfi_global_announcement/src/Plugin/Block/GlobalAnnouncementsBlock.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Drupal\helfi_global_announcement\Plugin\Block;
+
+use Drupal\Core\Cache\Cache;
+use Drupal\helfi_node_announcement\Plugin\Block\AnnouncementsBlockBase;
+use Drupal\helfi_node_announcement\Plugin\ExternalEntities\StorageClient\Announcements;
+use Drupal\node\Entity\Node;
+
+/**
+ * Provides 'global announcements' block.
+ *
+ * @Block(
+ *   id = "global_announcements",
+ *   admin_label = @Translation("Global announcements"),
+ * )
+ */
+class GlobalAnnouncementsBlock extends AnnouncementsBlockBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(): array {
+    $uuids = \Drupal::entityTypeManager()
+      ->getStorage('helfi_announcements')
+      ->getQuery()
+      ->execute();
+
+    $globalEntityStorage = $this->entityTypeManager->getStorage('helfi_announcements');
+    $cacheMaxAge = $globalEntityStorage->getExternalEntityType()->get('persistent_cache_max_age');
+
+    $externalAnnouncements = $globalEntityStorage
+      ->loadMultiple($uuids);
+
+    $announcementNodes = [];
+
+    foreach ($externalAnnouncements as $announcement) {
+      $linkUrl = NULL;
+      $linkText = NULL;
+      if ($announcement->hasField('announcement_link_text')) {
+        $linkText = $announcement->get('announcement_link_text')->value;
+        $linkUrl = $announcement->get('announcement_link_url')->value;
+      }
+
+      $announcementNodes[] = Node::create([
+        'type' => 'announcement',
+        'field_announcement_type' => $announcement->get('announcement_type')->value,
+        'field_announcement_link' => ['uri' => $linkUrl, 'title' => $linkText],
+        'langcode' => $announcement->get('langcode')->value,
+        'body' => strip_tags(html_entity_decode($announcement->get('body')->value)),
+        'title' => strip_tags(html_entity_decode($announcement->get('title')->value)),
```

Comment:
Use `Xss::filter` instead of native php functions: https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Utility%21Xss.php/function/Xss%3A%3Afilter/8.5.x